htlcswitch/interfaces.go

```diff
@@ -4,28 +4,9 @@ import (
 	"github.com/btcsuite/btcd/wire"
 	"github.com/lightningnetwork/lnd/channeldb"
 	"github.com/lightningnetwork/lnd/lnpeer"
-	"github.com/lightningnetwork/lnd/lntypes"
 	"github.com/lightningnetwork/lnd/lnwire"
 )
 
-// InvoiceDatabase is an interface which represents the persistent subsystem
```

Comment:
I don't think this move is warranted. As is, it allows the package to specify the requirement son the interface level it needs in order to carry out its duties. Keeping the interface in this package also allows us to avoid an additional package dependency as well.

Reply:
I moved it to be able to use a mock for hodl manager (otherwise a circ. dep would occur). As hodl manager is non existent now, the move could be undone.